src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -1157,7 +1157,7 @@ static Assembler::SEW elemtype_to_sew(BasicType etype) {
 
 #define patch_vtype(hsb, lsb, vlmul, vsew, vta, vma, vill)   \
     if (vill == 1) {                                         \
-      guarantee((vlmul | vsew | vta | vma == 0),             \
+      guarantee((vlmul | vsew | vta | vma) == 0,             \
```

Comment:
I see the `vill` parameter is always false in current code, which means this guarantee never gets excecuted. And I don't think we would make use of the `vill` field of vtype in future. So I personally perfer to remove this guarantee and its enclosing if block for now.

Reply:
Rather than removing the guarantee, wouldn't it be better to guarantee/assert `vill == 0`?
Although looking at uses, that argument is a bool, so it should be `guarantee(!vill, ...)`.

Reply:
Hi, Yes, that's better. Maybe: `guarantee(!vill, "should be");`

Reply:
I've changed the guarantee as discussed.  There are further cleanups possible
here, but I'll leave that to the riscv port maintainers.